bin/embeddings_manager.py

```diff
@@ -0,0 +1,235 @@
+
+
+def which():
+    for db_name, display_path in EmbeddingEnvironment.get_dict().items():
+        print(f"{db_name}:\t{display_path}")
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+
+    # Parent parser for selecting embeddings
+    selection_parser = ArgumentParser(add_help=False)
+    selection_parser.add_argument(
+        "embedding",
+        type=EmbeddingSelection.parse,
+        help="Embedding selection: <modelorg>/<model>/<database>/<version>"
+    )
+
+    # Subcommands
+    subparsers = parser.add_subparsers(required=True)
+    pull_parser = subparsers.add_parser(
+        "pull",
+        parents=[selection_parser],
+        help="Download embeddings",
+    )
+    pull_parser.set_defaults(func=pull)
+    use_parser = subparsers.add_parser(
+        "use",
+        parents=[selection_parser],
+        help="Set the active embeddings",
+    )
+    use_parser.set_defaults(func=use)
+    install_parser = subparsers.add_parser(
+        "install",
+        parents=[selection_parser],
+        help="Download and set the active embeddings (pull+use)",
+    )
+    install_parser.set_defaults(func=install)
+    make_parser = subparsers.add_parser(
+        "make",
+        parents=[selection_parser],
+        help="Generate embeddings",
+    )
+    make_parser.set_defaults(func=make)
+    push_parser = subparsers.add_parser(
+        "push",
+        parents=[selection_parser],
+        help="Upload embeddings",
+    )
+    push_parser.set_defaults(func=push)
+    rm_parser = subparsers.add_parser(
+        "rm",
+        parents=[selection_parser],
+        help="Remove specified embedding (locally)",
+    )
+    rm_parser.set_defaults(func=rm)
+    ls_parser = subparsers.add_parser(
+        "ls",
+        help="List locally installed embeddings",
+    )
+    ls_parser.set_defaults(func=ls)
+    ls_remote_parser = subparsers.add_parser(
+        "ls-remote",
+        help="List available embeddings on S3",
+    )
+    ls_remote_parser.set_defaults(func=ls_remote)
+    which_parser = subparsers.add_parser(
+        "which",
+        help="Reveal the current embeddings in use",
+    )
+    which_parser.set_defaults(func=which)
+
+    # Command-specific arguments
+    make_parser.add_argument(
+        "--openai-key",
+        help="API key for OpenAI"
+    )
+    make_parser.add_argument(
+        "--hf-key",
+        help="API key for HuggingFaceHub",
+    )
+    make_parser.add_argument(
+        "--device",
+        help="PyTorch device to use when running HuggingFace model locally [cpu/cuda]",
+    )
+    make_parser.add_argument(
+        "--neo4j-uri",
+        default="bolt://localhost:7687",
+        help="URI for Neo4j database connection",
+    )
+    make_parser.add_argument(
+        "--neo4j-username",
+        help="Username for Neo4j database connection",
+    )
+    make_parser.add_argument(
+        "--neo4j-password",
+        help="Password for Neo4j database connection",
+    )
```

Comment:
Are these Neo4j arguments actually needed?

Reply:
Well the neo4j docker container does not have a username or password. But the neo4j running on release does. I think we will eventually move to make the release server use the docker container but for now this is how it is setup.